sidecar/cook/sidecar/progress.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2020 Two Sigma Open Source, LLC
+#
+#  Permission is hereby granted, free of charge, to any person obtaining a copy
+#  of this software and associated documentation files (the "Software"), to
+#  deal in the Software without restriction, including without limitation the
+#  rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+#  sell copies of the Software, and to permit persons to whom the Software is
+#  furnished to do so, subject to the following conditions:
+#
+#  The above copyright notice and this permission notice shall be included in
+#  all copies or substantial portions of the Software.
+#
+#  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+#  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+#  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+#  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+#  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+#  FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+#  IN THE SOFTWARE.
+#
+"""Entrypoint for Cook's sidecar progress reporter."""
+
+import faulthandler
+import logging
+import os
+import requests
+import signal
+import sys
+
+import cook.sidecar.config as csc
+import cook.sidecar.tracker as cst
+from cook.sidecar.version import __version__
+
+
+def start_progress_trackers():
+    try:
+        config = csc.initialize_config(os.environ)
+
+        default_url = config.callback_url
+        current_url = default_url
+
+        def send_progress_message(message):
+            nonlocal current_url
+            try:
+                for i in range(config.max_post_attempts):
+                    response = requests.post(current_url, allow_redirects=False, timeout=config.max_post_time_secs, json=message)
+                    if response.status_code == 202:
+                        return True
```

Comment:
Please add a log that the POST was successful.

Nitpick: should we be particular about `202` or accept any `2XX` response?

Reply:
👍

Reply:
Changed to `if 200 <= response.status_code <= 299` :heavy_check_mark: